sqlconnect/internal/redshift/driver/dsn.go

```diff
@@ -53,8 +64,22 @@ func (cfg *RedshiftConfig) LoadOpts() []func(*config.LoadOptions) error {
 			cfg.SessionToken,
 		)))
 	}
+	if cfg.RoleARN != "" {
+		awsCfg, err := config.LoadDefaultConfig(ctx, config.WithRegion(cfg.Region))
```

Comment:
If an `AccessKeyID` and a `SecretAccessKey` have been provided in the config, shouldn't we use them for the sts config as well? Shall we always rely on environment variables for these?